File: frontend/src/content/core/contentRegistry.ts
import type { ContentPack } from './contentPack';

const packs = new Map<string, ContentPack>();

export function registerContentPack(pack: ContentPack) {
  packs.set(pack.packId, pack);
}

export function getContentPack(packId: string) {
  return packs.get(packId);
}

export function listContentPacks() {
  return [...packs.values()];
}

export function clearContentPacksForTest() {
  packs.clear();
}

File: backend/content/core/content_registry.py
from __future__ import annotations

from .content_pack import ContentPack

_PACKS: dict[str, ContentPack] = {}


def register_content_pack(pack: ContentPack) -> None:
    _PACKS[pack.packId] = pack


def get_content_pack(pack_id: str) -> ContentPack | None:
    return _PACKS.get(pack_id)


def list_content_packs() -> list[ContentPack]:
    return list(_PACKS.values())


def clear_content_packs_for_test() -> None:
    _PACKS.clear()
